spacy/util.py

```diff
@@ -1583,12 +1583,12 @@ def minibatch(items, size):
     so that batch-size can vary on each step.
     """
     if isinstance(size, int):
-        size_ = constant_schedule(size)
+        size_ = itertools.repeat(size)
     else:
-        size_ = size
+        size_ = iter(size)
     items = iter(items)
-    for step in itertools.count():
-        batch_size = size_(step)
+    while True:
+        batch_size = next(size_)
```

Comment:
General question: does the `Schedule` API (or the `Optimizer`) guarantee that any generator derived from it will never throw a `StopIteration` error? I see that we call `next` in the batcher code with that implicit expectation.

Reply:
Yes, that's guaranteed. It uses an infinite iterator:

https://github.com/explosion/thinc/blob/e58ca10bf97f8490eb4132208d80f3bb43431186/thinc/schedules.py#L62

Which is used to feed values to the schedule, which does not return `StopIteration` and must always return a value:

https://github.com/explosion/thinc/blob/e58ca10bf97f8490eb4132208d80f3bb43431186/thinc/schedules.py#L32